src/world_interface/real_world_interface.cpp

```diff
@@ -53,7 +53,7 @@ const kinematics::DiffDriveKinematics& driveKinematics() {
 
 namespace {
 // map that associates camera id to the camera object
-std::unordered_map<CameraID, std::shared_ptr<cam::Camera>> cameraMap;
+std::unordered_map<CameraID, std::weak_ptr<cam::Camera>> cameraMap;
```

Comment:
Maintain a weak pointer here so reference counting works. When the number of strong references (all held by client code) drops to 0 , the camera will be cleaned up.